src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -180,7 +180,8 @@ private BoundExpression VisitExpressionImpl(BoundExpression node)
             // statement means that this constraint is not violated).
             // Dynamic type will be erased in emit phase. It is considered equivalent to Object in lowered bound trees.
             Debug.Assert(visited == null || visited.HasErrors || ReferenceEquals(visited.Type, node.Type) ||
-                    visited.Type.Equals(node.Type, TypeCompareKind.IgnoreDynamicAndTupleNames));
+                    visited.Type.Equals(node.Type, TypeCompareKind.IgnoreDynamicAndTupleNames) ||
+                    node.Kind == BoundKind.DeconstructionAssignmentOperator);
```

Comment:
>node.Kind == BoundKind.DeconstructionAssignmentOperator [](start = 20, length = 55)

Please strengthen this condition by checking that IsUsed is false for this node.